lib/connection.js

```diff
@@ -328,15 +362,22 @@ class Connection extends EventEmitter {
     this.emit('_timeout');
   }
 
+  // Prepare a JSTP message to be sent over this connection
+  //   message - a message to prepare
+  //
+  _prepareMessage(message) {
+    return serde.stringify(message);
+  }
+
   // Send a JSTP message over this connection
-  //   message - a message to send
+  //   preparedMessage - a prepared message to send
+  //   message - a message object (used for development events thus optional)
   //
-  _send(message) {
-    const data = serde.stringify(message);
-    this.transport.send(data);
+  _send(preparedMessage, message) {
```

Comment:
I don't like the idea of creating the function just for stringifying messages, the output of which will always be parsed just again while sending the message, that doesn't make sense to me.